extensions/ui/org.eclipse.smarthome.ui.classic/src/main/java/org/eclipse/smarthome/ui/classic/internal/servlet/WebAppServlet.java

```diff
@@ -221,27 +228,32 @@ private boolean waitForChanges(EList<Widget> widgets) {
         Set<GenericItem> items = new HashSet<GenericItem>();
         if (itemRegistry != null) {
             for (Widget widget : widgets) {
-                String itemName = widget.getItem();
-                if (itemName != null) {
-                    try {
-                        Item item = itemRegistry.getItem(itemName);
-                        if (item instanceof GenericItem) {
-                            final GenericItem gItem = (GenericItem) item;
-                            items.add(gItem);
-                        }
-                    } catch (ItemNotFoundException e) {
-                        // ignore
-                    }
-                } else {
-                    if (widget instanceof Frame) {
-                        items.addAll(getAllItems(((Frame) widget).getChildren()));
-                    }
+                addItemWithName(items, widget.getItem());
+                if (widget instanceof Frame) {
+                    items.addAll(getAllItems(((Frame) widget).getChildren()));
+                }
+                for (VisibilityRule vr : widget.getVisibility()) {
+                    addItemWithName(items, vr.getItem());
                 }
             }
         }
         return items;
     }
 
+    private void addItemWithName(Set<GenericItem> items, String itemName) {
+        if (itemName != null) {
+            try {
+                Item item = itemRegistry.getItem(itemName);

```

Comment:
I know you just re-factored this code into another method, but I am asking myself why the `getItem()` method is called on the `itemRegistry` vs `itemUIregistry` in case of the basicui.